src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -133,36 +220,54 @@ def __init__(
             If `guest_start` or `guest_target` is defined but the other
             is not.
 
+        :class:`ValueError`
+            If the old API and new API are used simultaneously.
+
+        :class:`RuntimeError:
+            If no guests are provided.
+
         """
 
-        num_nones = sum(
-            1 for vector in (guest_start, guest_target)
-            if vector is None
-        )
-        if num_nones == 1:
-            raise TypeError(
-                'If guest_start or guest_target is defined, '
-                'the other must be too.'
+        if guests is not None and (
+            guest is not None or len({
+                displacement, guest_start, guest_target
+            }) > 1
+        ):
+            raise ValueError(
+                'You are attempting to use the old API with the new '
+                'API. Use the stk.host_guest.Guest API and `guests`.'
             )
 
-        if guest_start is None:
-            start = target = (1., 0., 0.)
+        if guests is not None:
+            building_block_vertices = self._get_vertices_from_guests(
+                host=host,
+                guests=guests,
+            )
+        elif guest is not None:
+            warnings.warn(
+                'You defined a Complex topology graph using the old '
+                'API (by defining `guest` and any optional arguments: '
+                '`guest_start`, `guest_target` and `displacement`). '
+                'This API will be removed from any version of stk '
+                'released on, or after, 01/01/22. Please use the '
+                '`guests` argument and the `stk.host_guest.Guest` '
+                'dataclass to define all necessary attributes.',
+                category=FutureWarning,
+            )
+            building_block_vertices = self._get_vertices_from_guest(
+                host=host,
+                guest=guest,
+                guest_start=guest_start,
+                guest_target=guest_target,
+                displacement=displacement,
+            )
         else:
-            start = guest_start = tuple(guest_start)
-            target = guest_target = tuple(guest_target)
-
-        # Save the values as None, for __repr__.
-        self._guest_start = guest_start
-        self._guest_target = guest_target
-        self._displacement = displacement
+            raise RuntimeError(
+                'The `guests` parameter must be provided.'
```

Comment:
Here, because it is a user facing error message, rather than markup used for documentation, quotes should be used, not backticks.

```
The "guests" parameter must be provided.
```

Reply:
Fixed throughout